pwndbg/aglib/kernel/rbtree.py

```diff
@@ -2,27 +2,29 @@
 
 from typing import Iterator
 
-import gdb
-
 import pwndbg
+import pwndbg.aglib.memory
+import pwndbg.aglib.typeinfo
+from pwndbg.aglib.kernel.macros import container_of
 from pwndbg.dbg import EventType
-from pwndbg.gdblib.kernel.macros import container_of
 
-rb_root_type: gdb.Type = None
-rb_node_type: gdb.Type = None
+rb_root_type: pwndbg.dbg_mod.Type = None
+rb_node_type: pwndbg.dbg_mod.Type = None
 
 
 @pwndbg.dbg.event_handler(EventType.NEW_MODULE)
 def init():
     global rb_root_type, rb_node_type
     try:
-        rb_root_type = gdb.lookup_type("struct rb_root")
-        rb_node_type = gdb.lookup_type("struct rb_node")
+        rb_root_type = pwndbg.aglib.typeinfo.load("struct rb_root")
+        rb_node_type = pwndbg.aglib.typeinfo.load("struct rb_node")
```

Comment:
After this change, there is no exception here that can be thrown

```py
In [1]: gdb.lookup_type('asdf')
---------------------------------------------------------------------------
error                                     Traceback (most recent call last)
Cell In[1], line 1
----> 1 gdb.lookup_type('asdf')

error: No type named asdf.

In [2]: pwndbg.aglib.typeinfo.load('asdf')

In [3]:
```

so the `try: ... except: ..` can be removed?

Reply:
I also wonder if `NEW_MODULE` is ever valid in the context of kernel debugging?

Reply:
probably `NEW_MODULE` is invalid in kernel context